components/modal/src/Modal.js

```diff
@@ -1,15 +1,35 @@
 /** @jsx jsx */
 
-import React, { Children, cloneElement, useState, useEffect } from 'react';
+import React, {
+	Children,
+	cloneElement,
+	createContext,
+	useContext,
+	useState,
+	useEffect,
+} from 'react';
 import { jsx } from '@westpac/core';
 import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import shortid from 'shortid';
 import FocusLock from 'react-focus-lock';
 import { CSSTransition } from 'react-transition-group';
-import { ModalBackdrop, StyledModal } from './styled';
-import { ModalHeader } from './ModalHeader';
-import { ModalBody } from './ModalBody';
+import { Backdrop, StyledModal } from './styled';
+
+// ==============================
+// Context and Consumer Hook
+// ==============================
+const ModalContext = createContext();
+
+export const useModalContext = () => {
+	const context = useContext(ModalContext);
+
+	if (!context) {
+		throw new Error('Modal inner components should be wrapped in a <Modal>.');
```

Comment:
Should make this message consistent

```suggestion
		throw new Error('Modal sub-components should be wrapped in a <Modal>.');
```

?